src/hb/paint_extents.rs

```diff
@@ -173,3 +177,54 @@ impl ttf_parser::colr::Painter<'_> for hb_paint_extents_context_t<'_> {
         self.pop_transform();
     }
 }
+
+trait TransformExt {
+    fn transform_distance(&self, dx: &mut f32, dy: &mut f32);
+    fn transform_point(&self, x: &mut f32, y: &mut f32);
+    fn transform_extents(&self, extents: &mut hb_extents_t);
+}
+
+impl TransformExt for Transform {
+    fn transform_distance(&self, dx: &mut f32, dy: &mut f32) {
+        let new_x = self.a * *dx + self.c * *dy;
+        let new_y = self.b * *dx + self.d * *dy;
+        *dx = new_x;
+        *dy = new_y;
+    }
+
+    fn transform_point(&self, x: &mut f32, y: &mut f32) {
+        self.transform_distance(x, y);
+        *x += self.e;
+        *y += self.f;
+    }
+
+    fn transform_extents(&self, extents: &mut hb_extents_t) {
+        let mut quad_x = [0.0f32; 4];
+        let mut quad_y = [0.0f32; 4];
+
+        quad_x[0] = extents.x_min;
+        quad_y[0] = extents.y_min;
+        quad_x[1] = extents.x_min;
+        quad_y[1] = extents.y_max;
+        quad_x[2] = extents.x_max;
+        quad_y[2] = extents.y_min;
+        quad_x[3] = extents.x_max;
+        quad_y[3] = extents.y_max;
+
+        for i in 0..4 {
```

Comment:
Maybe `zip()` will be faster and more readable?

Reply:
Hmm, I think in this case I would prefer to keep the code same as in harfbuzz, if that's okay.

Reply:
Sure.